test/src/main/java/org/axonframework/test/matchers/DeepEqualsMatcher.java

```diff
@@ -0,0 +1,146 @@
+            // Expected does not equal actual, and equals is implemented. Hence, we should not perform field equality.
+            noneMatchingEquals = true;
+            return false;
+        }
+        return matchingFields(expected.getClass(), expected, actual);
+    }
+
+    private boolean matchingTypes(Object actual) {
+        return expected.getClass().isInstance(actual) && expected.getClass().equals(actual.getClass());
+    }
+
+    private boolean matchingFields(Class<?> aClass, Object expectedValue, Object actual) {
+        boolean match = true;
+        for (Field field : aClass.getDeclaredFields()) {
+            if (filter.accept(field)) {
+                try {
+                    field.setAccessible(true);
+                    Object expectedFieldValue = field.get(expectedValue);
+                    Object actualFieldValue = field.get(actual);
+                    if (!Objects.deepEquals(expectedFieldValue, actualFieldValue)) {
+                        failedField = field;
+                        failedFieldExpected = expectedFieldValue;
+                        failedFieldActual = actualFieldValue;
+                        return false;
+                    }
+                } catch (Exception e) {
+                    if (e.getClass().getSimpleName().equals("InaccessibleObjectException")) {
+                        logger.warn("Could not confirm object field equality due to InaccessibleObjectException.");
+                        failedForAccessibilityException = true;
+                        return false;
+                    }
+                    throw new MatcherExecutionException("Could not confirm object equality due to an exception.", e);
+                }
+            }
+        }
+        if (aClass.getSuperclass() != Object.class) {
+            match = matchingFields(aClass.getSuperclass(), expectedValue, actual);
+        }
+        return match;
+    }
+
+    @Override
+    public void describeTo(Description description) {
+        description.appendText(expected.getClass().getName());
+        if (noneMatchingTypes) {
+            description.appendText(" does not match with the actual type.");
+        } else if (noneMatchingEquals) {
+            description.appendText(" does not equal with the actual instance.");
+        } else if (failedField != null) {
+            description.appendText(" (failed on field '")
+                       .appendText(failedField.getName())
+                       .appendText("').")
+                       .appendText(" Expected field value [")
+                       .appendValue(failedFieldExpected)
+                       .appendText("], but actual field value was [")
+                       .appendValue(failedFieldActual)
+                       .appendText("].");
+        } else if (failedForAccessibilityException) {
+            description.appendText(" failed during field equality with InaccessibleObjectException.");
```

Comment:
Could we expand this exception text to show why it is happening? Or as a comment in the original method when catching this exception and setting the boolean. Otherwise we can send people on a wild goose chase, while we already know why :)

Reply:
Good point! Will do.